go/tasks/plugins/hive/execution_state.go

```diff
@@ -143,14 +148,31 @@ func ConstructTaskLog(e ExecutionState) *idlCore.TaskLog {
 	}
 }
 
-func ConstructTaskInfo(e ExecutionState) *core.TaskInfo {
+func ConstructTaskInfo(tCtx core.TaskExecutionContext, e ExecutionState) *core.TaskInfo {
 	logs := make([]*idlCore.TaskLog, 0, 1)
 	t := time.Now()
+
+	metadata := &event.TaskExecutionMetadata{
+		GeneratedName: tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(),
```

Comment:
Following the changes to "if we know the value of the field, we shouldn't ask plugin authors to fill them in...", GeneratedName is something in our control...

And so is ResourcePoolInfo...

Reply:
oh wow! i got thrown off by the allocation token in the webapi plugin, ty for pointing this out re: resource manager